src/common/lib/util/multicaster.ts

```diff
@@ -1,25 +1,37 @@
+import { StandardCallback } from 'common/types/utils';
+import ErrorInfo from 'common/lib/types/errorinfo';
 import Logger from './logger';
 
-type AnyFunction = (...args: any[]) => unknown;
-
-export interface MulticasterInstance extends Function {
-  (...args: unknown[]): void;
-  push: (fn: AnyFunction) => void;
+export interface MulticasterInstance<T> extends Function {
+  (err?: ErrorInfo | null, result?: T): void;
+  push: (fn: StandardCallback<T>) => void;
+  /**
+   * Creates a promise that will be resolved or rejected when this instance is called.
+   */
+  createPromise: () => Promise<T>;
+  /**
+   * Syntatic sugar for when working in a context that uses promises; equivalent to calling as a function with arguments (null, result).
+   */
+  resolveAll(result: T): void;
+  /**
+   * Syntatic sugar for when working in a context that uses promises; equivalent to calling as a function with arguments (err).
+   */
+  rejectAll(err: ErrorInfo): void;
 }
 
-class Multicaster {
-  members: Array<AnyFunction>;
+class Multicaster<T> {
+  members: Array<StandardCallback<T>>;
 
   // Private constructor; use static Multicaster.create instead
-  private constructor(members?: Array<AnyFunction | undefined>) {
-    this.members = (members as Array<AnyFunction>) || [];
+  private constructor(members?: Array<StandardCallback<T> | undefined>) {
+    this.members = (members as Array<StandardCallback<T>>) || [];
   }
 
-  call(...args: unknown[]): void {
+  call(err?: ErrorInfo | null, result?: T): void {
```

Comment:
Is this still being used anywhere? If not I'd prefer marking it as `private` just to make it clear how it's intended to be used from now on

Reply:
Sure, added a58e93bc1539dfe808ef0f6a8cea32c98f047f99.